src/Invictus.Testing/LogicAuthentication.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+using System.Threading.Tasks;
+using GuardNet;
+using Microsoft.Azure.Management.Logic;
+using Microsoft.IdentityModel.Clients.ActiveDirectory;
+using Microsoft.Rest;
+
+namespace Invictus.Testing 
+{
+    /// <summary>
+    /// Authentication representation to authenticate with logic apps running on Azure.
+    /// </summary>
+    public class LogicAuthentication
+    {
+        private readonly Func<Task<LogicManagementClient>> _authenticateAsync;
+
+        private LogicAuthentication(Func<Task<LogicManagementClient>> authenticateAsync)
+        {
+            Guard.NotNull(authenticateAsync, nameof(authenticateAsync));
+
+            _authenticateAsync = authenticateAsync;
+        }
+
+        /// <summary>
+        /// Uses the service principal to authenticate with Azure.
+        /// </summary>
+        /// <param name="subscriptionId">The ID that identifies the subscription on Azure.</param>
+        /// <param name="tenantId">The ID where the resources are located on Azure.</param>
+        /// <param name="clientId">The ID of the client or application that has access to the logic apps running on Azure.</param>
+        /// <param name="clientSecret">The secret of the client or application that has access to the logic apps running on Azure.</param>
+        public static LogicAuthentication UsingServicePrincipal(string subscriptionId, string tenantId, string clientId, string clientSecret)
```

Comment:
```suggestion
        public static LogicAuthentication UsingServicePrincipal(string tenantId, string subscriptionId, string clientId, string clientSecret)
```

Reply:
Also created #30 to support for all Azure Clouds so that it's not blocking this PR

Reply:
Done signature suggestion.